lib/ocpp/v201/charge_point.cpp

```diff
@@ -192,6 +192,13 @@ void ChargePoint::disconnect_websocket(websocketpp::close::status::value code) {
 
 void ChargePoint::on_firmware_update_status_notification(int32_t request_id,
                                                          const FirmwareStatusEnum& firmware_update_status) {
+    if (this->firmware_status == firmware_update_status) {
+        if (request_id == -1 or
+            this->firmware_status_id.has_value() and this->firmware_status_id.value() == request_id) {
+            // already sent, do not send again
+            return;
+        }
+    }
```

Comment:
This checks if the request would not change the current status, and has the same request ID (i.e. refers to the same firmware update process) as the last one we received. What is the meaning of `request_id` being -1?

Reply:
Also, are we guaranteed to only ever have one firmware update process (i.e. only one valid `request_id`) valid at any time? This check wouldn't work if we're receiving updates for several concurrent update processes.

Reply:
The System module in everest-core currently ensures there's only one firmware update in progress at a given time, this seems to be the idea in OCPP 2.0.1 as well (see Figure 119 for the state machine of the firmware update which strongly suggests only one update at the time)